### PR TITLE
List and Offer Form

### PR DESCRIPTION
**Pull request by kespinola**
### Goal
- User can interact with listing and offer form.

### Changes
- Add react-router for doing nested routes within a next page.
- Build initial form for listing and offer.
- Header for NFT show.
Screen Shot 2022-02-07 at 6 37 39 AM

Screen Shot 2022-02-07 at 6 38 03 AM

### CODE REVIEW

**Review by echohtp — APPROVED** [review]
Great PR, learned a lot from this!

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -41,18 +44,32 @@ function App({ Component, pageProps }: AppProps) {
...
+    const handleRouteChange = (url: string) => {
+      navigate(url);
+    }
+
+    router.events.on('routeChangeStart', handleRouteChange)
+
+    return () => {
+      router.events.off('routeChangeStart', handleRouteChange)
+    }
+  }, []);
```
Comment (kespinola) [inline src/pages/_app.tsx:60]: This is keeping the router state in sync between next and react-router.

File: src/pages/nfts/[...address].tsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+          attributes {
+            traitType
+            value
+          }
+        }
+      }
+    `,
+    variables: {
+      subdomain: SUBDOMAIN,
+      address: ctx.params.address[0],
```
Comment (kespinola) [inline src/pages/nfts/[...address].tsx:48]: In order to get next serving the nest react router paths I had to convert to fuzzy param list that splits on path so the previous named address param is always in the first slot. react-router picks up the rest to do the actions area.

File: src/styles/globals.css [hunk 3]
```diff
@@ -26,6 +30,26 @@
...
+}
+.prefix-icon-sol:before {
+  content: "◎";
+  @apply inline-block absolute text-lg top-2 left-2 text-gray-500;
+}
+
+.wallet-adapter-button.wallet-adapter-button-trigger, .wallet-adapter-button:not([disabled]):hover {
+  @apply bg-white text-black hover:bg-gray-100 font-normal rounded-full h-14;
+
+}
```
Comment (kespinola) [inline src/styles/globals.css:49]: The modal buttons of picking a wallet provider get affected by this. Need to look at the user menu and picking provider according to the profile mocks.

**Pull request merged**